File: hello-kafka-unit-test/java/src/main/java/hellokafka/Consumer.java
package hellokafka;

import org.springframework.kafka.annotation.KafkaListener;
import org.springframework.stereotype.Service;

import java.util.ArrayList;
import java.util.List;
import java.util.concurrent.CountDownLatch;

@Service
public class Consumer {

    List<String> data = new ArrayList<>();
    CountDownLatch latch = new CountDownLatch(2);

    @KafkaListener(topics= {"hello-topic-ut"}, groupId = "kafka_java_id")
    void consume(String message) {
        System.out.println("RECEIVE message: $message");
        data.add(message);
        latch.countDown();
    }
}

File: hello-kafka-unit-test/src/main/kotlin/hellokafka/Consumer.kt
package hellokafka

import org.springframework.kafka.annotation.KafkaListener
import org.springframework.stereotype.Service
import java.util.concurrent.CountDownLatch

@Service
class Consumer {
    val data: MutableList<String> = ArrayList()
    var latch = CountDownLatch(2)

    @KafkaListener(topics= ["hello-topic-ut"], groupId = "kafka_kotlin_id")
    fun consume(message: String) {
        println("RECEIVE message: $message")
        data.add(message)
        latch.countDown()
    }
}
